### PR TITLE
New auto-update UX

### PR DESCRIPTION
**Pull request by goaaats**
Not my finest work but it'll do. One to-do left about opt-ins, but I'll try to fix that up later on.

### CODE REVIEW

File: Dalamud/Interface/Internal/DesignSystem/DalamudComponents.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Dalamud.Interface.Internal.DesignSystem;
+
+/// <summary>
+/// Private ImGui widgets for use inside Dalamud.
+/// </summary>
+internal static partial class DalamudComponents
+{
+}
```
Comment (KazWolfe) [inline Dalamud/Interface/Internal/DesignSystem/DalamudComponents.cs:8]: Intended empty file?

File: Dalamud/Interface/Internal/Windows/ChangelogWindow.cs [hunk 2]
```diff
@@ -286,7 +360,66 @@ void DrawNextButton(State nextState)
...
+                                                "Dalamud can update your plugins automatically, making sure that you always " +
+                                                "have the newest features and bug fixes. You can choose when and how auto-updates are run here."));
+                        ImGuiHelpers.ScaledDummy(2);
+                        
+                        ImGuiHelpers.SafeTextColoredWrapped(ImGuiColors.DalamudGrey, Loc.Localize("DalamudSettingsAutoUpdateDisclaimer1",
+                                                                "You can always update your plugins manually by clicking the update button in the plugin list. " +
+                                                                "You can also opt into updates for specific plugins by right-clicking them and selecting \"Always auto-update\"."));
+                        ImGuiHelpers.SafeTextColoredWrapped(ImGuiColors.DalamudGrey, Loc.Localize("DalamudSettingsAutoUpdateDisclaimer2",
+                                                                "Dalamud will never bother you about updates while you are not idle."));
```
Comment (KazWolfe) [inline Dalamud/Interface/Internal/Windows/ChangelogWindow.cs:387]:
```suggestion
                                                                "Dalamud will only bother you about updates when you're idle."));
```

No double-negative.

File: Dalamud/Interface/Internal/Windows/ChangelogWindow.cs [hunk 3]
```diff
@@ -384,4 +517,10 @@ bool CenteredIconButton(FontAwesomeIcon icon, string text)
     public void Dispose()
     {
     }
+
+    private static class FtueLevels
+    {
+        public const int Default = 1;
+        public const int AutoUpdateBehavior = 2;
+    }
```
Comment (KazWolfe) [inline Dalamud/Interface/Internal/Windows/ChangelogWindow.cs:525]: Not _particularly_ a fan of doing things this way - we should instead track whether the user has seen any given screen (at a specific version). Unfortunately I'm also lazy.

File: Dalamud/Interface/Internal/Windows/Settings/Tabs/SettingsTabAutoUpdate.cs [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        ImGuiHelpers.SafeTextColoredWrapped(ImGuiColors.DalamudWhite, Loc.Localize("DalamudSettingsAutoUpdateHint",
+                                                "Dalamud can update your plugins automatically, making sure that you always " +
+                                                "have the newest features and bug fixes. You can choose when and how auto-updates are run here."));
+        ImGuiHelpers.ScaledDummy(2);
+        
+        ImGuiHelpers.SafeTextColoredWrapped(ImGuiColors.DalamudGrey, Loc.Localize("DalamudSettingsAutoUpdateDisclaimer1",
+                                                "You can always update your plugins manually by clicking the update button in the plugin list. " +
+                                                "You can also opt into updates for specific plugins by right-clicking them and selecting \"Always auto-update\"."));
+        ImGuiHelpers.SafeTextColoredWrapped(ImGuiColors.DalamudGrey, Loc.Localize("DalamudSettingsAutoUpdateDisclaimer2",
+                                                "Dalamud will never bother you about updates while you are not idle."));
```
Comment (KazWolfe) [inline Dalamud/Interface/Internal/Windows/Settings/Tabs/SettingsTabAutoUpdate.cs:44]: Ditto to above.

File: Dalamud/Interface/Internal/Windows/Settings/Tabs/SettingsTabAutoUpdate.cs [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+                                                "This will override the settings above for the selected plugins."));
+
+        if (this.autoUpdatePreferences.Count == 0)
+        {
+            ImGuiHelpers.ScaledDummy(20);
+            
+            using (ImRaii.PushColor(ImGuiCol.Text, ImGuiColors.DalamudGrey))
+            {
+                ImGuiHelpers.CenteredText(Loc.Localize("DalamudSettingsAutoUpdateOptedInHint2",
+                                                       "You did not override auto-updates for any plugins yet."));
```
Comment (KazWolfe) [inline Dalamud/Interface/Internal/Windows/Settings/Tabs/SettingsTabAutoUpdate.cs:90]:
```suggestion
                                                       "You don't have auto-update rules for any plugins."));
```

File: Dalamud/Interface/Internal/DesignSystem/DalamudComponents.cs (hunk 1 above)
Comment (goaaats) [reply Dalamud/Interface/Internal/DesignSystem/DalamudComponents.cs:8]: Yeah, we do it the same way for the other components class.

**Review by KazWolfe — APPROVED** [review]

**Pull request merged**